### PR TITLE
Feature/alg 1132 od md

### PR DESCRIPTION
**Pull request by yurkovak**

### CODE REVIEW

File: documentation/source/ObjectDetection.md [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+                                       undo_preprocessing_func=my_undo_image_preprocessing)
+```
+The function you pass as `undo_preprocessing_func` will define how to undo dataset transforms and return the image back into its initial formal (BGR, uint8).
+This also allows you to test the correctness of your dataset implementation, since it saves images after they go through transforms. This may be especially useful for a train set with heavy augmentation transforms. You can see both the predictions and the ground truth, and give the ground truth box the desired opacity. 
+
+The saved train image for a dataset with a mosaic transform should look something like this:
+
+![train_24](images/train_24.jpg)
+
+## How to connect your own dataset
```
Comment (ofrimasad) [inline documentation/source/ObjectDetection.md:121]: before this. please add a code example of how everything you wrote above falls into place.
training params with the metric, the callback, the data-loaders

File: documentation/source/ObjectDetection.md [hunk 2]
```diff
@@ -0,0 +1,312 @@
+# Object Detection
+
+Object detection is a core task in computer vision that allows to detect and classify bounding boxes in images.
```
Comment (ofrimasad) [inline documentation/source/ObjectDetection.md:3]: let's make this intro reacher. maybe with GPT...

**Review by ofrimasad — COMMENTED** [review]
beautiful tutorial.
two minor comments inline

File: documentation/source/ObjectDetection.md [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+### Prerequisites
+
+1. You have to install SuperGradients first. Please refer to the [Installation](https://docs.deci.ai/super-gradients/documentation/source/installation/) section for more details.
+2. Prepare the COCO dataset as described in the [Computer Vision Datasets Setup](https://docs.deci.ai/super-gradients/src/super_gradients/training/datasets/Dataset_Setup_Instructions/) under Detection Datasets section. 
+
+After you meet the prerequisites, you can start training the model by running from the root of the repository:
+
+```bash
+python src/super_gradients/examples/train_from_recipe_example/train_from_recipe.py --config-name=coco2017_yolox multi_gpu=Off num_gpus=1
+```
```
Comment (Louis-Dupont) [inline documentation/source/ObjectDetection.md:27]: This is currently under the `### Prerequisites` but I think that it should have its own section, something like `### Training from recipe`, `### Quick start`, or something better

**Review by Louis-Dupont — COMMENTED** [review]
Looks really great! I just left a minor comment

File: documentation/source/ObjectDetection.md [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+
+def my_undo_image_preprocessing(im_tensor: torch.Tensor) -> np.ndarray:
+    im_np = im_tensor.cpu().numpy()
+    im_np = im_np[:, ::-1, :, :].transpose(0, 2, 3, 1)
+    im_np *= 255.0
+    return np.ascontiguousarray(im_np, dtype=np.uint8)
+
+
+model = models.get("yolox_s", pretrained_weights="coco", num_classes=80)
+imgs, targets = iter(train_dataloader).__next__()
```
Comment (BloodAxe) [inline documentation/source/ObjectDetection.md:108]: Is there a difference between `next(iter(train_dataloader))` and `__next__()`?
If there is none, then I suggest to use the `next` variant as it's more common.

Comment (yurkovak) [reply documentation/source/ObjectDetection.md]: Good points, looks much better, thanks 6e116f9

File: documentation/source/ObjectDetection.md (hunk 2 above)
Comment (yurkovak) [reply documentation/source/ObjectDetection.md:3]: I'm out of requests in chatGPT, hoping I'm poetic enough without it :D 6e116f9

File: documentation/source/ObjectDetection.md (hunk 1 above)
Comment (yurkovak) [reply documentation/source/ObjectDetection.md:227]: Added more story in 2653fbb

File: documentation/source/ObjectDetection.md [hunk 5]
```diff
@@ -0,0 +1,419 @@
+# Object Detection
+
+Object detection is a core task in computer vision that allows to detect and classify bounding boxes in images. 
+It's been gaining popularity and ubiquity extremely fast since the first breakthroughs in Deep Learning and advanced a wide range of companies, including the medical domain, surveillance, smart shopping, etc.
+It comes as no surprise considering that it covers two basic needs in an end-to-end manner: to find all present objects and to assign a class to each one of them, 
+while cleverly dealing with the background and its dominance over all other classes. 
+
+Due to this, most recent research publications dedicated to object detection focus on a good train-off between accuracy and speed.
```
Comment (ofrimasad) [inline documentation/source/ObjectDetection.md]: train-off -> trade off

**Review by ofrimasad — APPROVED** [review]
LGTM
